channels/channel.streams/slam/chn_slam.py

```diff
@@ -0,0 +1,202 @@
+                            "?ttag=PLAYER%3ANOPREROLL&tdsdk=js-2.9"
+                            "&pname=TDSdk&pversion=2.9&banners=none")
+        slam_fm.media_type = mediatype.AUDIO
+        slam_fm.isLive = True
+        slam_fm.add_stream(slam_fm.url)
+        slam_fm.complete = True
+        items.append(slam_fm)
+
+        slam_fm = MediaItem("Slam! Juize", "https://22543.live.streamtheworld.com/WEB09_MP3_SC"
+                            "?ttag=PLAYER%3ANOPREROLL&tdsdk=js-2.9"
+                            "&pname=TDSdk&pversion=2.9&banners=none")
+        slam_fm.media_type = mediatype.AUDIO
+        slam_fm.isLive = True
+        slam_fm.add_stream(slam_fm.url)
+        slam_fm.complete = True
+        items.append(slam_fm)
+
+        slam_fm = MediaItem("Slam! MixMarathon", "https://stream.slam.nl/web13_mp3"
+                            "?ttag=PLAYER%3ANOPREROLL&tdsdk=js-2.9"
+                            "&pname=TDSdk&pversion=2.9&banners=none")
+        slam_fm.media_type = mediatype.AUDIO
+        slam_fm.isLive = True
+        slam_fm.add_stream(slam_fm.url)
+        slam_fm.complete = True
+        items.append(slam_fm)
+
+        slam_fm = MediaItem("Slam! Hardstyle", "https://22323.live.streamtheworld.com/WEB11_MP3_SC"
+                            "?ttag=PLAYER%3ANOPREROLL&tdsdk=js-2.9"
+                            "&pname=TDSdk&pversion=2.9&banners=none")
+        slam_fm.media_type = mediatype.AUDIO
+        slam_fm.isLive = True
+        slam_fm.add_stream(slam_fm.url)
+        slam_fm.complete = True
+        items.append(slam_fm)
+
+        slam_fm = MediaItem("Slam! Housuh in de Pauzuh", "http://22553.live.streamtheworld.com/SLAM_MP3_SC"
+                            "?ttag=PLAYER%3ANOPREROLL&tdsdk=js-2.9"
+                            "&pname=TDSdk&pversion=2.9&banners=none")
+        slam_fm.media_type = mediatype.AUDIO
+        slam_fm.isLive = True
+        slam_fm.add_stream(slam_fm.url)
+        slam_fm.complete = True
+        items.append(slam_fm)
+        return data, items
+
+    def update_live_stream_m3u8(self, item):
+        """ Updates an existing MediaItem with more data.
+
+        Used to update none complete MediaItems (self.complete = False). This
+        could include opening the item's URL to fetch more data and then process that
+        data or retrieve it's real media-URL.
+
+        The method should at least:
+        * cache the thumbnail to disk (use self.noImage if no thumb is available).
+        * set at least one MediaStream.
+        * set self.complete = True.
+
+        if the returned item does not have a MediaSteam then the self.complete flag
+        will automatically be set back to False.
+
+        :param MediaItem item: the original MediaItem that needs updating.
+
+        :return: The original item with more data added to it's properties.
+        :rtype: MediaItem
+
+        """
+
+        for s, b in M3u8.set_input_stream_addon_input(item.url):
```

Comment:
It does not work like this. But like this: 

```
stream = item.add_stream(stream_url, 0)
M3u8.set_input_stream_addon_input(stream, self.headers)
item.complete = True
```

Reply:
What i did now?